scraper/src/github-scraper/discussion.ts

```diff
@@ -0,0 +1,132 @@
+import { octokit } from "./config.js";
+import { Discussion, Edge } from "./types.js";
+
+// Query to fetch discussions from GitHub
+const query = `
+query($org: String!, $cursor: String) {
+  organization(login: $org) {
+    repositories(first: 100, after: $cursor) {
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      edges {
+        node {
+          name
+          discussions(first: 100) {
+            edges {
+              node {
+                id
+                title
+                author {
+                  login
+                  avatarUrl
+                }
+                url
+                category{
+                  id
+                  name
+                  emoji
+                }
+                upvoteCount
+                reactions {
+                  totalCount
+                }
+                comments(first: 10) {
+                  edges {
+                    node {
+                      author {
+                        login
+                        avatarUrl
+                      }
+                      upvoteCount
+                      isAnswer
+                    }
+                  }
+                }
+                createdAt
+                isAnswered
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+`;
+async function fetchDiscussionsForOrg(org: string, cursor = null) {
+  const variables = { org, cursor };
+  const response = await octokit.graphql(query, variables);
+  const discussions = response.organization.repositories.edges.map(
+    (edge: Edge) => edge.node.discussions.edges,
+  );
+  const hasNextPage = response.organization.repositories.pageInfo.hasNextPage;
+  const nextCursor = response.organization.repositories.pageInfo.endCursor;
+
+  const allDiscussions = discussions.flat();
+
+  if (hasNextPage) {
+    const nextDiscussions: Discussion[] = await fetchDiscussionsForOrg(
+      org,
+      nextCursor,
+    );
+    return allDiscussions.concat(nextDiscussions);
+  }
```

Comment:
use `octokit.graphql.paginate` instead of `octokit.graphql`, so that pagination is handled by octokit itself.